runtime/nativeMain/src/kotlinx/benchmark/NativeBlackhole.kt

```diff
@@ -1,13 +1,37 @@
 package kotlinx.benchmark
 
 actual class Blackhole {
-    actual fun consume(obj: Any?) {}
-    actual fun consume(bool: Boolean) {}
-    actual fun consume(c: Char) {}
-    actual fun consume(b: Byte) {}
-    actual fun consume(s: Short) {}
-    actual fun consume(i: Int) {}
-    actual fun consume(l: Long) {}
-    actual fun consume(f: Float) {}
-    actual fun consume(d: Double) {}
+    @ThreadLocal
+    companion object {
+        var consumer = 0
+    }
+    actual fun consume(obj: Any?) {
+        // hashCode now is implemented as taking address of object, so it's suitable now.
+        // If implementation is changed `Blackhole` should be reimplemented.
+        consumer += obj.hashCode()
```

Comment:
That's a virtual call that also may be way too heavy for a user-supplied class. That's just a time bomb waiting to explode, so I'd suggest using Blackhole approach (non-ordered volatile write, LLVM directive, w/e).


Also, see https://bugs.openjdk.java.net/browse/JDK-8252505 for the potential pitfalls